lwt/client/dns_client_lwt.ml

```diff
@@ -26,10 +26,7 @@ module Transport : Dns_client.S
     mutable fd : [ `Plain of Lwt_unix.file_descr | `Tls of Tls_lwt.Unix.t ] option ;
     mutable connected_condition : (unit, [ `Msg of string ]) result Lwt_condition.t option ;
     mutable requests : (Cstruct.t * (Cstruct.t, [ `Msg of string ]) result Lwt_condition.t) IM.t ;
-    mutable he : Happy_eyeballs.t ;
-    mutable cancel_connecting : (int * unit Lwt.u) list Happy_eyeballs.Waiter_map.t;
-    mutable waiters : ((Ipaddr.t * int) * Lwt_unix.file_descr, [ `Msg of string ]) result Lwt.u Happy_eyeballs.Waiter_map.t ;
-    timer_condition : unit Lwt_condition.t ;
+    he : Happy_eyeballs_lwt.t ;
```

Comment:
I'm not sure to understand why we need here an other `Happy_eyeballs.t` where the given `H.t` has its own `Happy_eyeballs.t` too. Is it expected?

Reply:
comes from efb53e325904cd7916acc4c6629b611ef24e0c37 -- I guess we can remove it from the record here, since `type stack = Happy_eyeballs_lwt.t`...

Reply:
Yes, but in this case, we must extend `Happy_eyeballs_mirage.S` to be able to get the `Happy_eyeballs.t` from the `Happy_eyeballs_mirage.t` instance.